File: aluminium/src/main/java/de/zalando/demo/aluminium/ArtistController.java
package de.zalando.demo.aluminium;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;

@RestController
public class ArtistController {

    @GetMapping("/artists/{artistId}")
    public Artist getArtist(@PathVariable String artistId) {

        switch (artistId) {
            case "elvis":
                return new Artist("elvis",
                        "Elvis A. Presley",
                        LocalDate.of(1935, 1, 8)
                );
        }

        throw new ArtistNotFoundException("Artist [" + artistId + "]");
    }

}

File: 75_zalando_playground/iron/src/main/kotlin/de/roskenet/iron/ArtistController.kt
package de.roskenet.iron

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

data class Artist(val name: String, val genre: String)

@RestController
class ArtistController {

    @GetMapping("/api/artists")
    fun getArtists(): List<Artist> = listOf(
        Artist(name = "Elvis Presley", genre = "Rock'n'roll"),
        Artist(name = "Amy Winehouse", genre = "Soul")
    )
}